### ISSUE TITLE
`reportUninitializedInstanceVariable` false positive when initialized in `__new__`

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#455]
```py
class Foo:
    a: int # error: reportUninitializedInstanceVariable
    def __new__(cls):
        self = super().__new__(cls)
        self.a = 1
        return self
```
